lib/compiler_rt/x86_cpu_model.zig

```diff
@@ -0,0 +1,428 @@
+            break :blk .amd;
+        },
+        else => .other,
+    };
+    cpu.features[0] = features[0];
+    cpu_extra_features = features[1..].*;
+
+    return 0;
+}
+
+const Model = extern struct {
+    vendor: Vendor = .unknown,
+    type: Type = .unknown,
+    subtype: Subtype = .unknown,
+    features: [1]u32 = .{0},
+};
+
+const FeatureSet = [4]u32;
+comptime {
+    inline for (@typeInfo(Feature).Enum.fields) |f| {
+        if (f.value >= @bitSizeOf(FeatureSet))
+            @compileError(@import("std").fmt.comptimePrint("Feature.{s} ({}) bitindex too large", .{ f.name, f.value }));
+    }
+}
+
+fn hasFeature(set: FeatureSet, f: Feature) bool {
+    const f_value = @intFromEnum(f);
+    const bit = set[f_value / 32] & (@as(u32, 1) << @as(u5, @intCast(f_value % 32)));
+    return bit != 0;
+}
+fn setFeature(set: *FeatureSet, f: Feature) void {
+    const f_value = @intFromEnum(f);
+    set[f_value / 32] |= (@as(u32, 1) << @as(u5, @intCast(f_value % 32)));
+}
+
+// These values match the values in LLVM.
+
+const Vendor = enum(u32) {
+    unknown = 0,
+    intel,
+    amd,
+    other,
+};
+const Type = enum(u32) {
+    unknown = 0,
+    intel_bonnell,
+    intel_core2,
+    intel_corei7,
+    amdfam10h,
+    amdfam15h,
+    intel_silvermont,
+    intel_knl,
+    amd_btver1,
+    amd_btver2,
+    amdfam17h,
+    intel_knm,
+    intel_goldmont,
+    intel_goldmont_plus,
+    intel_tremont,
+    amdfam19h,
+    zhaoxin_fam7h,
+    intel_sierraforest,
+    intel_grandridge,
+    intel_clearwaterforest,
+};
+const Subtype = enum(u32) {
+    unknown = 0,
+    intel_corei7_nehalem,
+    intel_corei7_westmere,
+    intel_corei7_sandybridge,
+    amdfam10h_barcelona,
+    amdfam10h_shanghai,
+    amdfam10h_istanbul,
+    amdfam15h_bdver1,
+    amdfam15h_bdver2,
+    amdfam15h_bdver3,
+    amdfam15h_bdver4,
+    amdfam17h_znver1,
+    intel_corei7_ivybridge,
+    intel_corei7_haswell,
+    intel_corei7_broadwell,
+    intel_corei7_skylake,
+    intel_corei7_skylake_avx512,
+    intel_corei7_cannonlake,
+    intel_corei7_icelake_client,
+    intel_corei7_icelake_server,
+    amdfam17h_znver2,
+    intel_corei7_cascadelake,
+    intel_corei7_tigerlake,
+    intel_corei7_cooperlake,
+    intel_corei7_sapphirerapids,
+    intel_corei7_alderlake,
+    amdfam19h_znver3,
+    intel_corei7_rocketlake,
+    zhaoxin_fam7h_lujiazui,
+    amdfam19h_znver4,
+    intel_corei7_graniterapids,
+    intel_corei7_graniterapids_d,
+    intel_corei7_arrowlake,
+    intel_corei7_arrowlake_s,
+    intel_corei7_pantherlake,
+};
+const Feature = enum(u32) {
+    cmov = 0,
+    mmx,
+    popcnt,
+    sse,
+    sse2,
+    sse3,
+    ssse3,
+    sse4_1,
+    sse4_2,
+    avx,
+    avx2,
+    sse4a,
+    fma4,
+    xop,
+    fma,
+    avx512f,
+    bmi,
+    bmi2,
+    aes,
+    pclmul,
+    avx512vl,
+    avx512bw,
+    avx512dq,
+    avx512cd,
+    avx512er,
+    avx512pf,
+    avx512vbmi,
+    avx512ifma,
+    avx5124vnniw, // TODO: detect this in std/zig/system/x86.zig
```

Comment:
The general approach is to eliminate TODOs to make the context explicit to readers and simplify external contributions etc.
If you don't want to make this in this PR or a follow-up, then please make a very brief text what would be needed so another person can dive in quickly.